File: ios/FLEXR/UI/Navigation/Destinations.swift
import SwiftUI

/// Die vier Hauptbereiche der Tab-Leiste.
enum TopLevelDestination: String, CaseIterable, Identifiable, Hashable {
    case swipe, matches, chats, account

    var id: String { rawValue }

    /// Beschriftung als Textschluessel — aufgeloest wird erst dort, wo
    /// gezeichnet wird; nur da ist die gewaehlte Sprache bekannt.
    var labelKey: L {
        switch self {
        case .swipe: .navSwipe
        case .matches: .navMatches
        case .chats: .navChats
        case .account: .navAccount
        }
    }

    var icon: FlexrGlyph.Kind {
        switch self {
        case .swipe: .dumbbell
        case .matches: .symbol(FlexrIcon.matches)
        case .chats: .symbol(FlexrIcon.chats)
        case .account: .symbol(FlexrIcon.account)
        }
    }
}

/// Ziele innerhalb eines Tabs. Typisierte Routen statt String-Bastelei an den
/// Aufrufstellen — die Entsprechung von `Routes` in der Android-App.
enum Route: Hashable {
    case chat(matchID: String)
    case matchProfile(matchID: String)
    case verification
    /// FLEXR Premium. Früher ein eigener Navigationsbaum (der gesperrte
    /// Zustand nach Ablauf des Probemonats), seit dem 10.09.2026 ein normales
    /// Ziel aus dem Kontobereich.
    case premium
    case legal(LegalDocument)
}

enum LegalDocument: String, CaseIterable, Identifiable, Hashable {
    case faq, impressum, datenschutz, agb, sicherheit, nutzungsrichtlinien, strafverfolgung

    var id: String { rawValue }

    /// Nur der Titel der Ansicht ist uebersetzt — der Inhalt in
    /// `UI/Legal/LegalContent.swift` bleibt bewusst auf Deutsch, weil er in
    /// dieser Fassung verbindlich ist.
    var titleKey: L {
        switch self {
        case .faq: .legalFaq
        case .impressum: .legalImpressum
        case .datenschutz: .legalDatenschutz
        case .agb: .legalAgb
        case .sicherheit: .legalSicherheit
        case .nutzungsrichtlinien: .legalNutzungsrichtlinien
        case .strafverfolgung: .legalStrafverfolgung
        }
    }
}

File: android-native/app/src/main/java/flexr/social/app/ui/navigation/Destinations.kt
package flexr.social.app.ui.navigation

import androidx.annotation.StringRes
import androidx.compose.ui.graphics.vector.ImageVector
import flexr.social.app.R
import flexr.social.app.core.designsystem.icon.FlexrIcons

/** Alle Ziele der App. Typisierte Routen statt String-Bastelei an den Aufrufstellen. */
object Routes {
    const val LOGIN = "login"
    const val REGISTER = "register"
    const val PAYWALL = "paywall"

    const val SWIPE = "swipe"
    const val MATCHES = "matches"
    const val CHATS = "chats"
    const val ACCOUNT = "account"

    const val MATCH_PROFILE = "matchProfile/{matchId}"
    const val CHAT = "chat/{matchId}?origin={origin}"
    const val VERIFICATION = "verification"

    /** Alters- und Identitätsprüfung eines noch nicht freigeschalteten Kontos. */
    const val VERIFICATION_GATE = "verificationGate"
    const val VERIFICATION_DOCUMENT = "verificationDocument"

    const val LEGAL = "legal/{document}"

    fun matchProfile(matchId: String) = "matchProfile/$matchId"
    fun chat(matchId: String, origin: String = CHATS) = "chat/$matchId?origin=$origin"
    fun legal(document: LegalDocument) = "legal/${document.name}"
}

/**
 * Die Rechtstexte. Nur der Titel der Ansicht ist uebersetzt - der Inhalt in
 * [flexr.social.app.ui.legal.LegalContent] bleibt bewusst auf Deutsch, weil er
 * in dieser Fassung verbindlich ist.
 */
enum class LegalDocument(@StringRes val titleRes: Int) {
    FAQ(R.string.legal_faq),
    IMPRESSUM(R.string.legal_impressum),
    DATENSCHUTZ(R.string.legal_datenschutz),
    AGB(R.string.legal_agb),
    SICHERHEIT(R.string.legal_sicherheit),
    NUTZUNGSRICHTLINIEN(R.string.legal_nutzungsrichtlinien),
    STRAFVERFOLGUNG(R.string.legal_strafverfolgung),
}

/** Die vier Hauptbereiche der unteren Navigation. */
enum class TopLevelDestination(
    val route: String,
    @StringRes val labelRes: Int,
    val icon: ImageVector,
) {
    SWIPE(Routes.SWIPE, R.string.nav_swipe, FlexrIcons.Swipe),
    MATCHES(Routes.MATCHES, R.string.nav_matches, FlexrIcons.Matches),
    CHATS(Routes.CHATS, R.string.nav_chats, FlexrIcons.Chats),
    ACCOUNT(Routes.ACCOUNT, R.string.nav_account, FlexrIcons.Account),
}
